test/ctia/http/middleware/ratelimit_test.clj

```diff
@@ -0,0 +1,155 @@
+
+(def reset-redis
+  (fn [t]
+    (car/wcar {} {} (car/flushdb))
+    (t)))
+
+(defn apply-fixtures
+  [properties fn]
+  (let [fixture-fn
+        (join-fixtures [reset-redis
+                        helpers/fixture-properties:clean
+                        (helpers/fixture-properties:static-auth "user" "pwd")
+                        #(helpers/with-properties properties (%))
+                        es-helpers/fixture-properties:es-store
+                        helpers/fixture-ctia
+                        es-helpers/fixture-delete-store-indexes])]
+    (fixture-fn fn)))
+
+(deftest rate-limit-test
+  (with-redefs [time/now (constantly (time/date-time 2017 02 16 0 0 0))]
+    (let [call #(helpers/get "ctia/status"
+                             :headers {"Authorization" (str "Bearer " jwt-token)
+                                       "Origin" "http://external.cisco.com"})]
+      (testing "Group limit"
+        (apply-fixtures
+         ["ctia.http.rate-limit.default-group-limit" 5
+          "ctia.http.rate-limit.enabled" true]
+         (fn []
+           (let [response (call)]
+             (testing "Rate limit headers"
+               (is (= 200 (:status response)))
+               (is (= "5" (get-in response [:headers "X-RateLimit-GROUP-Limit"]))))
```

Comment:
if `X-RateLimit-GROUP-Remaining` is also in header I would test its presence here.

Reply:
`X-RateLimit-GROUP-Remaining` is not returned